pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisConsumer.java

```diff
@@ -150,8 +150,8 @@ private BytesStreamMessage extractStreamMessage(Record record, String shardId) {
     String sequenceNumber = record.sequenceNumber();
     KinesisPartitionGroupOffset offset = new KinesisPartitionGroupOffset(shardId, sequenceNumber);
     // NOTE: Use the same offset as next offset because the consumer starts consuming AFTER the start sequence number.
-    StreamMessageMetadata.Builder builder =
-        new StreamMessageMetadata.Builder().setRecordIngestionTimeMs(timestamp).setOffset(offset, offset);
+    StreamMessageMetadata.Builder builder = new StreamMessageMetadata.Builder().setRecordIngestionTimeMs(timestamp)
+        .setSerializedValueSize(record.data().asByteArray().length).setOffset(offset, offset);
```

Comment:
`record.data().asByteArray().length`
can this be null?

Reply:
`record.data().asByteArray().length` does not require additional null checks. 

Several lines above, there's this line of source code: 
```
byte[] value = record.data().asByteArray();
```

I have modified the change to 
```
.setSerializedValueSize(value.length)
```

Reply:
sounds good, let's change it to value.length.